substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -1156,17 +1150,21 @@ impl<T: Config> Pallet<T> {
         };
 
         // Handle grace
-        let contract = Self::handle_grace(&mut contract, usable_balance, amount_due)?;
+        let contract = Self::handle_grace(
+            &mut contract,
+            usable_balance,
+            contract_lock.amount_locked + amount_due,
```

Comment:
Ok, so here you take into account the accumulated amount due to evaluate if user stays in Grace or not
I ll create a new variable for readability
`total_amount_due` or `accumulated_amount_due`
and write some comment to explain why

// While in grace period the contract amount locked continue to increase
// and need to be taken into account for at each grace period evaluation

Reply:
:+1: